### PR TITLE
docs: update sitemap's version history section

### PR DESCRIPTION
**Pull request by Lada496**
### What?
This is an update on the documentation at the sitemap version history to add the update (https://github.com/vercel/next.js/pull/53765) in [v14.2.0](https://github.com/vercel/next.js/releases/tag/v14.2.0).
### Why?
I really appreciate that the next sitemap generator supports localization. It would be great if we could see which version or higher provides this support. 

### How?
I added a new row at the top of the version history table.

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 4e1c83ec40a5b1d2708199979cd38488f42744d8

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 4e1c83ec40a5b1d2708199979cd38488f42744d8

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Review by samcx — APPROVED** [review]
Thank you for submitting a PR!

<picture data-single-emoji=":lgtm:" title=":lgtm:">:lgtm:</picture>

File: docs/02-app/02-api-reference/02-file-conventions/01-metadata/sitemap.mdx [hunk 1]
```diff
@@ -295,5 +295,6 @@ type Sitemap = Array<{
 
 | Version   | Changes                                                      |
 | --------- | ------------------------------------------------------------ |
+| `v14.2.0` | Add support for localizations in sitemap generator.          |
```
Comment (huozhi) [inline docs/02-app/02-api-reference/02-file-conventions/01-metadata/sitemap.mdx:298]: reword it a little bit without saying generator
```suggestion
| `v14.2.0` | Add localizations support to sitemaps.          |
```

**Review by samcx — APPROVED** [review]
Thank you for submitting a PR!

<picture data-single-emoji=":lgtm:" title=":lgtm:">:lgtm:</picture>

**Review by huozhi — APPROVED** [review]

**Pull request merged**